### PR TITLE
boot: copy kernel/base to data partition in makeBootable20RunMode

### PR DESCRIPTION
**Pull request by mvo5**
The kernel/base must be copied into the ubuntu-data partition
when making the system runnable. This commit adds the needed
code to do this.

### CODE REVIEW

File: boot/boot.go [hunk 1]
```diff
@@ -424,6 +425,19 @@ func makeBootable20RunMode(model *asserts.Model, rootdir string, bootWith *Boota
...
 
+	// copy kernel/base into the ubuntu-date partition
+	ubuntuDataMnt := filepath.Join(runMnt, "ubuntu-data")
+	snapBlobDir := dirs.SnapBlobDirUnder(filepath.Join(ubuntuDataMnt, "system-data"))
+	if err := os.MkdirAll(snapBlobDir, 0755); err != nil {
+		return err
+	}
+	for _, fn := range []string{bootWith.BasePath, bootWith.KernelPath} {
+		dst := filepath.Join(snapBlobDir, filepath.Base(fn))
+		if err := osutil.CopyFile(fn, dst, 0); err != nil {
```
Comment (pedronis) [inline boot/boot.go]: do we want either of CopyFlagPreserveAll|CopyFlagSync here ?  or do our own sync on the dir afterward?

**Comment by mvo5** [comment]
There is one catch here - we need to make sure that /run/mnt/ubuntu-data is actually the real "ubuntu-data" partitio and not our tmpfs. We could use e.g. "/run/snapd-inst/mnt/ubuntu-data" or similar. We will need the same for ubuntu-boot later, i.e. it must be mounted after it got created.

**Comment by mvo5** [comment]
I.e. we need something like https://github.com/snapcore/snapd/pull/7923 first before this PR is useful.

File: boot/boot.go [hunk 2]
```diff
@@ -424,6 +425,19 @@ func makeBootable20RunMode(model *asserts.Model, rootdir string, bootWith *Boota
 	// - create grub.cfg instead of using the gadget one
 	// - extract kernel
 
+	// copy kernel/base into the ubuntu-date partition
```
Comment (bboozzoo) [inline boot/boot.go]:
```suggestion
	// copy kernel/base into the ubuntu-data partition
```

**Pull request merged**